olp-cpp-sdk-dataservice-read/src/generated/model/SubscribeResponse.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+
+#include <boost/optional.hpp>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+namespace model {
+
+class SubscribeResponse final {
+ public:
+  const boost::optional<std::string>& GetNodeBaseURL() const {
+    return node_base_url_;
+  }
+  void SetNodeBaseURL(std::string value) { node_base_url_ = std::move(value); }
+
+  const boost::optional<std::string>& GetSubscriptionId() const {
+    return subscription_id_;
+  }
+  void SetSubscriptionId(std::string value) {
+    subscription_id_ = std::move(value);
+  }
+
+  const boost::optional<std::string>& GetCorrelationId() const {
+    return x_correlation_id;
+  }
+  void SetCorrelationId(std::string value) {
+    x_correlation_id = std::move(value);
+  }
+
+ private:
+  /// A subscription/Kafka Consumer is created in a specific server instance.
+  /// This base URL represents the specific server node on which this
+  /// subscription/Kafka Consumer was created. This base URL should be used
+  /// instead of the base URL returned by the `lookup` service when making
+  /// requests to these endpoints: /partitions, /offsets, /seek and /subscribe
+  /// (DELETE).
+  boost::optional<std::string> node_base_url_;
+
+  /// A unique identifier of the subscription. It is used to make requests to
+  /// these endpoints: /partitions, /offsets, /seek and /subscribe (DELETE).
+  boost::optional<std::string> subscription_id_;
+
+  /// A tracing ID (correlation-id) to use in subsequent requests. This ID
+  /// should be obtained from HTTP response header.
+  boost::optional<std::string> x_correlation_id;
```

Comment:
I dont think we need to add this here. This represent more the body response and should not contain anything else. The X-Correlation-Id should be a separate output to the StreamApi::Subscribe() method.

Reply:
We discussed it during regarding Stream Layer POC. I thought we agreed to add this field here.
Will add one more parameter to StreamApi::Subscribe() for X-Correlation-Id output.

Reply:
Not sure we agreed on this. It doesn't look to me like it is part of the response.
I would rather prefer to put X-Correlation-Id separate as it is more work needed to extract it.